### PR TITLE
Setup github actions and goreleaser

### LINKED ISSUE #3: Add CI builds
A CI system should be setup to perform builds of dnsbench for releases. [Travis](https://travis-ci.org/) may be a good option.

### PR DESCRIPTION
**Pull request by richardmarshall**
Adds github actions workflows to run basic checks on PRs and merges to main as well as for release on tag pushes.

Includes a number of updates to the goreleaser config and Makefile to improve the results of build / test and release flow.

Resolves #3

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
```
Comment (benco5) [inline .github/workflows/ci.yaml]: @richardmarshall Should actually be `main`?

File: .goreleaser.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      - "--label=org.opencontainers.image.title={{ .ProjectName }}"
+      - "--label=org.opencontainers.image.vendor=AskMediaGroup"
+      - "--label=org.opencontainers.image.description=dnsbench {{ .Version }}"
+      - "--label=org.opencontainers.image.url=https://github.com/askmediagroup/dnsbench"
+      - "--label=org.opencontainers.image.documentation=https://github.com/askmediagroup/dnsbench/blob/main/README.md"
+      - "--label=org.opencontainers.image.source=https://github.com/askmediagroup/dnsbench"
+      - "--label=org.opencontainers.image.licenses=Apache-2.0"
+      - "--label=org.opencontainers.image.version={{ .Version }}"
+      - "--label=org.opencontainers.image.revision={{ .FullCommit }}"
+      - "--label=org.opencontainers.image.created={{ time \"2006-01-02T15:04:05Z07:00\" }}"
```
Comment (benco5) [inline .goreleaser.yaml:52]: @richardmarshall What's up with that hardcoded timestamp string?

**Review by benco5 — COMMENTED** [review]
Few questions.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (richardmarshall) [reply .github/workflows/ci.yaml]: Yes! Thank you for catching that.

File: .goreleaser.yaml (hunk 2 above)
Comment (richardmarshall) [reply .goreleaser.yaml:52]: It's a template, the goreleaser date function accepts a timestamp as an argument and parses it to figure out how you want the actual timestamp to be formatted.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (richardmarshall) [reply .github/workflows/ci.yaml]: fixed

**Review by benco5 — APPROVED** [review]
Saw your replies / change, so otherwise LGTM

**Pull request merged**